### ISSUE TITLE
Linux kernel indentation buggy reports 

### ISSUE THREAD

**Issue by jerome-benoit** [hebaishi/linter-gcc#140]
## Checklist
* [x] Check that you have the GNU C Compiler installed by typing ```gcc --version``` into a terminal.
* [x] If you don't have the ```linter``` package installed, either update to the latest version of ```linter-gcc``` or install linter directly with ```apm install linter```.
* [x] Ensure that you are using a valid grammar - either C/C++/C++14
* [] Ensure that you have read all the relevant information in the [wiki](https://github.com/hebaishi/linter-gcc/wiki), particularly the [Common Issues](https://github.com/hebaishi/linter-gcc/wiki/Common-Issues) page.

## Issue (including debugging messages):

linter-gcc report wrong errors and warnings on C file with Linux kernel indentation style. 
To reproduce, test linter-gcc with this file http://www.mcs.anl.gov/~kazutomo/list/list.h. 

Thks.
